neqo-http3/src/frames/hframe.rs

```diff
@@ -150,7 +150,7 @@ impl HFrame {
                 update_frame.encode_varint(*element_id);
 
                 let mut priority_enc: Vec<u8> = Vec::new();
-                write!(priority_enc, "{priority}").unwrap();
+                write!(priority_enc, "{priority}").expect("write OK");
```

Comment:
This might be a good case for [`encode_vvec_with`](https://github.com/mozilla/neqo/blob/bcb89cddc68474f73947d65a823cb0458e0fc682/neqo-common/src/codec.rs#L387)

Making a temporary buffer is pretty heavyweight.  Implementing `Write` for `Encoder` is more work, but it's neater and pretty easy.

Reply:
I opened https://github.com/mozilla/neqo/issues/2652. Suggest to merge this one as-is, then change it there.

(We do seem to do a lot of `Vec::new` in the http3 code...)